src/expressions.md

```diff
@@ -88,9 +88,26 @@ created and used instead. The lifetime of temporary values is typically the
 innermost enclosing statement; the tail expression of a block is considered
 part of the statement that encloses the block.
 
-When a temporary rvalue is being created that is assigned into a `let`
-declaration, however, the temporary is created with the lifetime of the
-enclosing block instead, as using the enclosing statement (the `let`
+A first exception is when a temporary value is created in the
```

Comment:
I'd argue that this is not an exception - the only exception is the `let` rule. It's just that the conditional of an `if` or `while` are counted as a single statement for the purposes of the rule.

Also, side note, you don't have to do the whole taking a reference thing in order to create a temporary object. If a pvalue is not assigned to an lvalue, or is not used to construct an object, then a temporary object is automatically created.